pkg/util/azureclient/mgmt/authorization/denyassignment_addons.go

```diff
@@ -30,3 +38,50 @@ func (c *denyAssignmentClient) ListForResourceGroup(ctx context.Context, resourc
 
 	return result, nil
 }
+
+func (c *denyAssignmentClient) DeleteDenyAssignment(ctx context.Context, fpTokenCredential *azidentity.ClientCertificateCredential, subscriptionDoc *api.SubscriptionDocument, doc *api.OpenShiftClusterDocument) error {
```

Comment:
minor: could just pass in the subscriptionDoc.ID directly instead of the entire document given it's the only thing from it which is used

Reply:
Same could apply to doc for creating:
managedResourceGroupName := strings.Split(doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, "/")[4]